### PR TITLE
feat(queue-events): add better event typing

### PR DESCRIPTION
**Pull request by roggervalf**
ref https://rjzaworski.com/2019/10/event-emitters-in-typescript

### CODE REVIEW

File: tests/test_script_loader.ts [hunk 1]
```diff
@@ -38,11 +38,6 @@ describe('scriptLoader', () => {
   });
 
   describe('when using path mappings', () => {
-    it('correctly determines the path to the project root', () => {
```
Comment (roggervalf) [inline tests/test_script_loader.ts:41]: this test is deleted because rootPath is a private property

File: src/classes/queue-events.ts [hunk 2]
```diff
@@ -9,88 +9,80 @@ import { DELAY_TIME_5 } from '../utils';
 import { QueueBase } from './queue-base';
 import { RedisConnection } from './redis-connection';
 
-export interface QueueEventsDeclaration {
+export interface QueueEventsList {
```
Comment (manast) [inline src/classes/queue-events.ts:12]: I would call this interface: "QueueEventsListener" which in my opinion represents better what this interface is.

File: src/classes/queue-events.ts [hunk 3]
```diff
@@ -99,90 +91,61 @@ export interface QueueEventsDeclaration {
...
-   * @param {'waiting-children'} event
-   * @callback listener
    */
-  on(
-    event: 'waiting-children',
-    listener: (args: { jobId: string }, id: string) => void,
+  'waiting-children': (args: { jobId: string }, id: string) => void;
+}
+
+export interface QueueEventsDeclaration {
```
Comment (manast) [inline src/classes/queue-events.ts:139]: Don't we also need a "off" or "removeEventListener" declaration?

File: src/classes/job.ts [hunk 4]
```diff
@@ -780,8 +780,8 @@ export class Job<
       const completedEvent = `completed:${jobId}`;
       const failedEvent = `failed:${jobId}`;
 
-      queueEvents.on(completedEvent, onCompleted);
-      queueEvents.on(failedEvent, onFailed);
+      queueEvents.on(completedEvent as any, onCompleted);
```
Comment (manast) [inline src/classes/job.ts:783]: This is not super nice but I guess there are not so many better options...

File: src/classes/queue-events.ts [hunk 5]
```diff
@@ -195,6 +158,29 @@ export interface QueueEventsDeclaration {
 export class QueueEvents extends QueueBase implements QueueEventsDeclaration {
   private running = false;
 
+  emit<U extends keyof QueueEventsList>(
```
Comment (manast) [inline src/classes/queue-events.ts:161]: If we add these methods, why do we need this declaration?

```QueueEventsDeclaration```

Comment (roggervalf) [reply src/classes/queue-events.ts]: yeah we don't need QueueEventsDeclaration in this case

File: src/classes/job.ts (hunk 4 above)
Comment (roggervalf) [reply src/classes/job.ts:783]: yep, as now we are using static events, there is no glob version for add these cases, I tried with [key: string] but it does not work either

**Review by manast — APPROVED** [review]

**Pull request merged**